test/hotspot/jtreg/compiler/loopopts/parallel_iv/TestParallelIvInIntCountedLoop.java

```diff
@@ -0,0 +1,290 @@
+            a += 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static int testIntCountedLoopWithIntIVWithRandomStrides(int stop) {
+        int a = 0;
+        for (int i = 0; i < stop; i += stride) {
+            a += stride2;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIV(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVZero(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += 0;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVMax(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += Long.MAX_VALUE;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVMaxMinusOne(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += Long.MAX_VALUE - 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVMaxPlusOne(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += Long.MAX_VALUE + 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVWithStrideTwo(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i += 2) {
+            a += 2;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVWithStrideMinusOne(int stop) {
+        long a = 0;
+        for (int i = stop; i > 0; i += -1) {
+            a += 1;
+        }
+
+        return a;
+    }
+
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static long testIntCountedLoopWithLongIVWithRandomStrides(int stop) {
+        long a = 0;
+        for (int i = 0; i < stop; i += stride) {
+            a += (long) stride2;
+        }
+
+        return a;
+    }
```

Comment:
I'm also missing some cases where both the `init` and `limit` are variable.
You could also sprinkle in a case or two where we use `<=` instead of `<`.